### ISSUE TITLE
Audio offset calibration doesn't take into account the offset in last play when quitting beatmap and playing it again without using "retry" button

### ISSUE THREAD

**Issue by serobor** [ppy/osu#28753]
### Type

Game behaviour

### Bug description

Audio offset calibration doesn't take in account  the offset in last play when quitting beatmap and playing it again without using "retry" button.
I was hitting early on purpose, so I got +50 ms offset calibration, on next play I got -4,46 ms. If I calibrated offset without quitting,  I would get 45,54ms offset. Instead I quit beatmap and started playing it again, calibrated offset, but  I got -4,46 ms. The 50ms which I got before, were ignored.

### Screenshots or videos

https://github.com/ppy/osu/assets/16684745/2eaca738-3bf5-4bae-a48b-948b557c0075

### Version

2024.625.2-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16116882/compressed-logs.zip)

**Comment by peppy** [comment]
If you wait a minute at song select before playing the second time does this work correctly?

**Comment by serobor** [comment]
> If you wait a minute at song select before playing the second time does this work correctly?

No, waiting doesn't help. After you quit beatmap and average hit error is saved (3 ms early for example), if you play beatmap again and calibrate audio offset, it will set 3ms, ignoring previous offset. It will work correctly only if you retry a beatmap without returning to song select (retrying from a result screen works correctly too)

**Comment by shiumano** [comment]
I don't think it's a delay, I think something is wrong.

https://github.com/user-attachments/assets/bd89da49-6d36-4bad-9757-64b446bfc5ba

**Comment by Sebaex** [comment]
From what i found is that `lastPlayBeatmapOffset` here goes to 0 once the player exits the song so it messes up on where to consider the auto offset, if there was another way to get the map's current offset it should fix the problem
https://github.com/ppy/osu/blob/c90e04092ad88cd6bd08237b951c0502a3e15b9c/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs#L214-L217
